samples/go/csv/schema.go

```diff
@@ -117,6 +117,103 @@ func GetSchema(r *csv.Reader, headers []string) KindSlice {
 	return so.MostSpecificKinds()
 }
 
+func GetFieldNamesFromIndices(headers []string, indices []int) []string {
+	result := make([]string, len(indices))
+	for i, idx := range indices {
+		result[i] = headers[idx]
+	}
+	return result
+}
+
+// combinations - n choose m combination without repeat - emit all possible `length` combinations from values
+func combinationsWithLength(values []int, length int, emit func([]int)) {
```

Comment:
I think it would be good to have a test for these more primitive methods, rather than just relying on running the program to test them.